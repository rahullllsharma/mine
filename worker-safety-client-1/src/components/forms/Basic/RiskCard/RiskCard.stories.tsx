import type { ComponentMeta, ComponentStory } from "@storybook/react";

import { RiskCard } from "./RiskCard";

export default {
  title: "JSB/RiskCard",
  component: RiskCard,
} as ComponentMeta<typeof RiskCard>;

export const Playground: ComponentStory<typeof RiskCard> = args => (
  <RiskCard {...args} />
);

Playground.args = {
  risk: "ArcFlash",
};
