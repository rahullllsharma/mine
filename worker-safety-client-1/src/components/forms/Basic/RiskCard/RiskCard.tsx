/* eslint-disable @next/next/no-img-element */
import type { CriticalRisk } from "@/api/codecs";
import classnames from "classnames";

export type RiskCardProps = {
  headerClassName?: string;
  className?: string;
  risk: CriticalRisk;
  onClick?: () => void;
};

const riskLabel = (risk: CriticalRisk): string => {
  switch (risk) {
    case "ArcFlash":
      return "Arc Flash";
    case "CollisionLossOfControl":
      return "Collision Loss of Control";
    case "ConfinedSpace":
      return "Enclosed or Confined Space";
    case "ExposureToEnergy":
      return "Exposure to Energy";
    case "FallOrFallArrest":
      return "Fall or Fall Arrest";
    case "FireOrExplosion":
      return "Fire or Explosion";
    case "HoistedLoads":
      return "Hoisted Loads";
    case "LineOfFire":
      return "Line of Fire";
    case "MobileEquipment":
      return "Mobile Equipment";
    case "TrenchingOrExcavation":
      return "Trenching or Excavation";
  }
};

const riskImageSrc = (risk: CriticalRisk): string => `/assets/JSB/${risk}.svg`;

const RiskCard: React.FC<RiskCardProps> = ({
  className,
  headerClassName,
  children,
  risk,
  onClick, // Added onClick prop to check the entire card
}) => {
  const handleClick = () => {
    if (onClick) {
      onClick(); // Call the onClick function passed from RiskToggleCard
    }
  };
  return (
    <div
      className={classnames(
        "max-w-[231px] shadow-10 rounded-lg cursor-pointer",
        className
      )}
      onClick={handleClick} // Attached onClick handler to the check or uncheck the card
    >
      <div className={classnames("bg-white px-4 py-2", headerClassName)}>
        <span className="text-base font-semibold">{riskLabel(risk)}</span>
      </div>

      <div className="flex flex-col items-center p-4 bg-white">
        <img className="pb-2" src={riskImageSrc(risk)} alt="Risk image" />
        <span className="pb-2 text-sm font-semibold">
          Critical risk observed
        </span>
        {children}
      </div>
    </div>
  );
};

export { RiskCard };
