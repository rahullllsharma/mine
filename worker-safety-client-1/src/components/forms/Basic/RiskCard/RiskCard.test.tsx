import type { RiskCardProps } from "./RiskCard";
import { render } from "@testing-library/react";

import { RiskCard } from "./RiskCard";

describe(RiskCard.name, () => {
  const props: RiskCardProps = {
    risk: "ArcFlash",
  };

  it("renders correctly", () => {
    const { asFragment } = render(<RiskCard {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
