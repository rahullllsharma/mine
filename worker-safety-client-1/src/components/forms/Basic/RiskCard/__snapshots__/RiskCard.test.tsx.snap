// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RiskCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="max-w-[231px] shadow-10 rounded-lg cursor-pointer"
  >
    <div
      class="bg-white px-4 py-2"
    >
      <span
        class="text-base font-semibold"
      >
        Arc Flash
      </span>
    </div>
    <div
      class="flex flex-col items-center p-4 bg-white"
    >
      <img
        alt="Risk image"
        class="pb-2"
        src="/assets/JSB/ArcFlash.svg"
      />
      <span
        class="pb-2 text-sm font-semibold"
      >
        Critical risk observed
      </span>
    </div>
  </div>
</DocumentFragment>
`;
